### PR TITLE
[Android] Add Android platform support

### PR DESCRIPTION
**Pull request by hangedfish**
# Details
* support Android Bionic-libc, for android 6.0+.
* add Android build script.
* add Android build tutorial.

### CODE REVIEW

File: docs/book/zh/src/os/android.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    34
+    $ ../wasmedge --reactor factorial.wasm fac 12
+    479001600
+    $
+    $ cd js
+    $ ./../wasmedge --dir .:. qjs.wasm hello.js 1 2 3
+    Hello 1 2 3
+    ```
+
+
```
Comment (hydai) [inline docs/book/zh/src/os/android.md:58]: Please remove the redundant empty lines.

**Review by hydai — CHANGES_REQUESTED** [review]

File: utils/android/build_for_android.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+WASMEDGE_ROOT_PATH=$(dirname $(dirname $(dirname $0)))
+
+cd ${WASMEDGE_ROOT_PATH}
+mkdir build
+cd build
+
+echo $(pwd)
+
+if ! cmake .. -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_BUILD_AOT_RUNTIME=OFF -DCMAKE_SYSTEM_NAME=Android -DCMAKE_SYSTEM_VERSION=23 -DCMAKE_ANDROID_ARCH_ABI=arm64-v8a -DCMAKE_ANDROID_NDK=$ANDROID_NDK_HOME -DCMAKE_ANDROID_STL_TYPE=c++_static -DWASMEDGE_BUILD_AOT_RUNTIME=OFF; then
```
Comment (O3Ol) [inline utils/android/build_for_android.sh:18]: Please delete the `-DWASMEDGE_BUILD_AOT_RUNTIME=OFF` at the end of the line.

Besides, `CMAKE_SYSTEM_VERSION` should be set by user.

**Review by O3Ol — COMMENTED** [review]
It could work.

File: utils/android/build_for_android.sh (hunk 2 above)
Comment (hangedfish) [reply utils/android/build_for_android.sh]: The minimum Android API level requirement for WasmEdge is 23, only using `seekdir` func in bionic libc. Using a higher API level reduces system compatibility and provides no additional benefits.

**Comment by hydai** [comment]
Hi @hangedfish
Please clean up your commit history, it looks like combine some extra commits which are not created by you.

File: cmake/Helper.cmake [hunk 3]
```diff
@@ -6,7 +6,7 @@ if(CMAKE_BUILD_TYPE STREQUAL Release OR CMAKE_BUILD_TYPE STREQUAL RelWithDebInfo
   if(NOT WASMEDGE_FORCE_DISABLE_LTO)
     set(WASMEDGE_INTERPROCEDURAL_OPTIMIZATION ON)
   endif()
-  if (CMAKE_GENERATOR STREQUAL Ninja)
+  if (CMAKE_GENERATOR STREQUAL Ninja AND NOT ANDROID)
```
Comment (hydai) [inline cmake/Helper.cmake:9]: You don't need to add extra check, because you are not using `ninja` building system.

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by hangedfish** [comment]
This PR cannot be applied to production environment. Due to Android 10+ SELinux policy constraints, the application's system call exec() does no works on files within the application home directory and `/data/local/temp`. Close.

**Comment by juntao** [comment]
Hmm, I actually think it is still worthwhile. Commercial android smartphones are likely to be locked down and they will not allow an alternative runtime. But Android is used in many IoT or customize devices, I think those are the primary targets for WasmEdge's Android support.

**Pull request closed without merging**